### PR TITLE
python38Packages.eth-hash: fix the build by not depending on safe-pysha3

### PR DESCRIPTION
**Pull request by ivan**
eth-hash works with both pycryptodome and safe-pysha3 (https://github.com/ethereum/eth-hash/issues/46#issuecomment-1314029211); pycryptodome builds on Python 3.8, while safe-pysha3 requires Python 3.9+.

This fixes the build of grab-site, which is still stuck on Python 3.8, and depends on eth-hash through its autobahn dependency.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc maintainer @SuperSandro2000
cc @nevivurn @wegank

### CODE REVIEW

**Review by nevivurn — APPROVED** [review]
LGTM, also maybe safe-pysha3 should be fixed to say `pythonOlder "3.9"` instead of `pythonOlder "3.8"`, as it is broken for < 3.9, not < 3.8.

**Comment by SuperSandro2000** [comment]
Shouldn't this then trigger a rebuild for grab-site ?

@ofborg build grab-site

**Comment by SuperSandro2000** [comment]
thats why there is no rebuild ``error: matplotlib-3.7.0 not supported for interpreter python3.8``

**Comment by ivan** [comment]
@SuperSandro2000 good catch, sorry about that. I forgot I had another commit to fix matplotlib; added now.

File: pkgs/development/python-modules/matplotlib/default.nix [hunk 1]
```diff
@@ -80,7 +80,7 @@ buildPythonPackage rec {
   pname = "matplotlib";
   format = "pyproject";
 
-  disabled = pythonOlder "3.9";
+  disabled = pythonOlder "3.8";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/matplotlib/default.nix:83]: thats correct https://github.com/matplotlib/matplotlib/blob/v3.7.0/setup.py#L296

**Pull request merged**